shuf: Do not read input when -n0 is given

BenWiederhake opened the pull request:
This is explicitly tested by some suites, including the GNU test suite.

Together with #5980 and #5989, this should make uutils pass the GNU test suite for shuf. Yay! :D

sylvestre commented:
please run rustfmt :)

cakebaker commented on tests/by-util/test_shuf.rs:
```diff
@@ -147,6 +147,81 @@ fn test_head_count() {
+        .arg("-n0")
+        .pipe_in(vec![])
+        .succeeds();
+    result.no_stderr();
+
+    let result_seq: &str = result.stdout_str();
+    assert!(result_seq.is_empty(), "Output must be completely empty, not even newline");
```
You can simplify it to:
```suggestion
    result.no_output();
```

BenWiederhake commented:
Changes since last push:
- Added "[unwritable](https://en.wiktionary.org/wiki/unwritable)" to the cspell-ignore header. 
- `test_cp::test_cp_arg_interactive_update` flaked. No action taken.

cakebaker commented on tests/by-util/test_shuf.rs:
```diff
@@ -147,6 +149,81 @@ fn test_head_count() {
+#[test]
+fn test_zero_head_count_pipe() {
+    let result = new_ucmd!().arg("-n0").pipe_in(vec![]).succeeds();
+    // Output must be completely empty, not even a newline!
```
The comment is redundant
```suggestion
```

BenWiederhake replied:
I disagree: Without the comment there's the danger that someone might think it's a bug, or unintentional.

cakebaker commented on tests/by-util/test_shuf.rs:
```diff
@@ -147,6 +149,81 @@ fn test_head_count() {
+#[test]
+fn test_zero_head_count_pipe_explicit() {
+    let result = new_ucmd!().arg("-n0").arg("-").pipe_in(vec![]).succeeds();
+    result.no_stderr();
+
+    let result_seq: &str = result.stdout_str();
+    assert!(
+        result_seq.is_empty(),
+        "Output must be completely empty, not even newline"
+    );
```
It can be simplified here, and in the other test functions, to:
```suggestion
    result.no_output();
```

cakebaker commented on tests/by-util/test_shuf.rs:
```diff
@@ -147,6 +149,81 @@ fn test_head_count() {
+    );
+}
+
+#[test]
+fn test_zero_head_count_file_touch_output() {
+    new_ucmd!()
+        .arg("-n0")
+        .arg("-o")
+        .arg("/invalid/unwritable")
+        .pipe_in(vec![])
+        .fails()
+        .stderr_contains("failed to open '/invalid/unwritable' for writing:");
+}
```
There is a test function missing similar to this function to ensure the success case is working (i.e. the specified file is created).

BenWiederhake replied:
You're right, thanks!

BenWiederhake commented:
Changes since last push:
- Rebased on current main
- Simplified a few tests by calling `.no_output()`. Thanks @cakebaker!
- Split `test_zero_head_count_file_touch_output` into three tests: Negative (tries to write to unwritable path), positive (new file created), another positive (touches existing file). Thanks @cakebaker!

cakebaker commented:
Good job, thanks!
